src/python/dxpy/bindings/dxfile_functions.py

```diff
@@ -123,6 +124,72 @@ def download_dxfile(dxid, filename, chunksize=dxfile.DEFAULT_BUFFER_SIZE, append
                                    show_progress=show_progress, project=project, **kwargs)
 
 
+# Check if a program (wget, curl, etc.) is on the path, and
+# can be called.
+def _which(program):
+    def is_exe(fpath):
+        return os.path.isfile(fpath) and os.access(fpath, os.X_OK)
+
+    for path in os.environ["PATH"].split(os.pathsep):
+        exe_file = os.path.join(path, program)
+        if is_exe(exe_file):
+            return exe_file
+    return None
+
+# Caluclate the md5 checkum for [filename], and raise
+# an exception if the checksum is wrong.
+def _verify(filename, md5digest):
+    md5sum_exe = _which("md5sum")
+    if md5sum_exe is None:
+        err_exit("md5sum is not installed on this system")
+    cmd = [md5sum_exe, "-b", filename]
+    try:
+        print("Calculating checksum")
+        cmd_out = subprocess.check_output(cmd)
+    except subprocess.CalledProcessError:
+        err_exit("Failed to run md5sum: " + str(cmd))
+
+    line = cmd_out.strip().split()
+    if len(line) != 2:
+        err_exit("md5sum returned weird results: " + str(line))
+    actual_md5 = line[0]
+    if actual_md5 != md5digest:
+        err_exit("Checksum doesn't match " + actual_md5 + "  expected:" + md5digest)
```

Comment:
I'd reword it to "Checksum " + actual_md5 + " doesn't match expected: " + md5digest